local/rest_api_gcbm/app.py

```diff
@@ -177,71 +183,419 @@ def gcbm_upload():
-                ) as mof:
-                    json.dump(modules_output, mof)
-            else:
-                # Save file immediately
-                file.save(f"{os.getcwd()}/input/{project_dir}/{file.filename}")
+    # input files follow a strict structure
+    if not os.path.exists(f"{input_dir}/disturbances"):
+        os.makedirs(f"{input_dir}/disturbances")
+    if not os.path.exists(f"{input_dir}/classifiers"):
+        os.makedirs(f"{input_dir}/classifiers")
+    if not os.path.exists(f"{input_dir}/db"):
+        os.makedirs(f"{input_dir}/db")
+    if not os.path.exists(f"{input_dir}/miscellaneous"):
+        os.makedirs(f"{input_dir}/miscellaneous")
+
+    # store files following structure defined in curl.md
+    if "disturbances" in request.files:
+        for file in request.files.getlist("disturbances"):
+            file.save(f"{input_dir}/disturbances/{file.filename}")
     else:
         return {"error": "Missing configuration file"}, 400
 
-    # Save input
-    if "input" in request.files:
-        for file in request.files.getlist("input"):
-            # Save file immediately
-            file.save(f"{os.getcwd()}/input/{project_dir}/{file.filename}")
+    if "classifiers" in request.files:
+        for file in request.files.getlist("classifiers"):
+            file.save(f"{input_dir}/classifiers/{file.filename}")
     else:
-        return {"error": "Missing input"}, 400
+        return {"error": "Missing configuration file"}, 400
 
-    # Save db
     if "db" in request.files:
         for file in request.files.getlist("db"):
-            # Save file immediately
-            file.save(f"{os.getcwd()}/input/{project_dir}/{file.filename}")
+            file.save(f"{input_dir}/db/{file.filename}")
     else:
-        return {"error": "Missing database"}, 400
+        return {"error": "Missing configuration file"}, 400
+
+    if "miscellaneous" in request.files:
+        for file in request.files.getlist("miscellaneous"):
+            file.save(f"{input_dir}/miscellaneous/{file.filename}")
+    else:
+        return {"error": "Missing configuration file"}, 400
+
+    get_config_templates(input_dir)
+    get_modules_cbm_config(input_dir)
+    get_provider_config(input_dir)
 
     return {
-        "data": "All files uploaded sucessfully. Proceed to the next step of the API at gcbm/dynamic."
-    }, 200
+        "data": "All files uploaded succesfully. Proceed to the next step of the API at gcbm/dynamic."
+    }
+
+
+def get_config_templates(input_dir):
+    if not os.path.exists(f"{input_dir}/templates"):
+        shutil.copytree(
+            f"{os.getcwd()}/templates", f"{input_dir}/templates", dirs_exist_ok=False
+        )
+
+
+# TODO: there needs to be a link between the files configured here append
+# the ["vars"] attribute of modules_cbm.json -> CBMDisturbanceListener
+# current hack is to drop the last five characters, but thats very fragile
```

Comment:
@padmajabhol - has this been resolved? It looks to me like you've emptied the `vars` template (which is great!) and maybe we don't need to drop `_moja` any more?

Can you please also make sure you test what happens if we do not upload any disturbances?

Reply:
I have emptied the `vars` since we don't need to hardcode it prior to the pre-processing, it takes up all the file names present in the disturbances folder in the required format, same with `provider_config`. 
We care currently splitting the `disturbances_20xx_moja.tiff` to get just `disturbances_20xx_moja` and then taking away the last 5 characters `_moja` which leaves us with `disturbances_20xx`. Are you saying we don't need the names in this format?

I'll test the run without uploading any disturbance files.

Reply:
I don't think so. Originally I dropped `_moja` to match the modules_cbm template, so if that hack is resolved then I think it can probably take whatever the filename is (without the extension). 🤞

Reply:
Got, I'll change it back.